lib/YaST/Firstboot/LanguageAndKeyboardPage.pm

```diff
@@ -0,0 +1,52 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The class introduces all accessing methods for
+# firstboot Language and Keyboard client
+
+package YaST::Firstboot::LanguageAndKeyboardPage;
+use parent 'YaST::Firstboot::GenericPage';
+use strict;
+use warnings;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init();
+}
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
+    $self->{cb_keyboard} = $self->{app}->combobox({id => 'keyboard'});
+    $self->{cb_language} = $self->{app}->combobox({id => 'language'});
+    $self->{lb_main}     = $self->{app}->label({debug_label => 'Language and Keyboard Layout'});
+    return $self;
+}
+
+sub is_shown {
+    my ($self) = @_;
+    return $self->{lb_main}->exist();
+}
+
+sub check_selected_language {
+    my ($self, $expected_value) = @_;
+    my $selected_language = $self->{cb_language}->value();
+    die "The selected language \"$selected_language\" is not the expected one \"$expected_value\"" unless $expected_value eq $selected_language;
+}
+
+sub check_selected_keyboard {
+    my ($self, $expected_value) = @_;
+    my $selected_language = $self->{cb_keyboard}->value();
+    die "The selected keyboard language \"$selected_language\" is not the expected one \"$expected_value\"" unless $expected_value eq $selected_language;
```

Comment:
Same point as above, we should not do asserts in the page object, but just provide interfaces to manipulate the controls.